ee/hogai/tool.py

```diff
@@ -117,3 +117,73 @@ def format_system_prompt_injection(self, context: dict[str, Any]) -> str:
             key: (json.dumps(value) if isinstance(value, dict | list) else value) for key, value in context.items()
         }
         return self.root_system_prompt_template.format(**formatted_context)
+
+
+# Define the exact possible page keys for navigation. Extracted using the following Cursor prompt:
+# "
+# List every key of the `frontend/src/products.tsx::productUrls` object, whose function takes either zero arguments,
+# or only optional arguments, of which the first is named `tab` (exactly `tab`, and it CAN be the only argument).
+# Exclude beta or alpha products. Exclude scenes about importing a file. Your only output should be a list of those
+# string keys in Python `Literal[..., ..., ...]` syntax. Once done, verify whether indeed each item of the output
+# satisfies the argument criterion.
+# "
+PageKeyLiterals = Literal[
```

Comment:
IMO, this should be defined in the frontend, and then accessed in here through an automatically generated `Literal` via our schema gen

Reply:
+1 to this